test/city-council-district/city-council-district.e2e-spec.ts

```diff
@@ -104,4 +107,108 @@ describe("City Council District e2e", () => {
       expect(response.body.error).toBe(HttpName.INTERNAL_SEVER_ERROR);
     });
   });
+
+  describe("findCapitalProjectsByCityCouncilDistrictId", () => {
+    const mock =
+      cityCouncilDistrictRepositoryMock.checkCityCouncilDistrictByIdMocks[0];
+
+    it("should 200 amd return all capital projects for a city council district", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects`)
+        .expect(200);
+      expect(() =>
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+      const parsedBody =
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        );
+      expect(parsedBody.limit).toBe(20);
+      expect(parsedBody.offset).toBe(0);
+    });
+
+    it("should 200 amd return all capital projects for a city council district with user specified offset and limit", async () => {
+      const limit = 10;
+      const offset = 3;
+      const response = await request(app.getHttpServer())
+        .get(
+          `/city-council-districts/${mock.id}/capital-projects?limit=${limit}&offset=${offset}`,
+        )
+        .expect(200);
+      expect(() =>
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+      const parsedBody =
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        );
+      expect(parsedBody.limit).toBe(limit);
+      expect(parsedBody.offset).toBe(offset);
+    });
+
+    it("should 400 and when finding by an invalid id", async () => {
+      const invalidId = "MN";
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${invalidId}/capital-projects`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
```

Comment:
I don't have a good reason for this. But, for 404s, the name of the error is in the "message" and not the "error"

```suggestion
      expect(response.body.message).toBe(HttpName.BAD_REQUEST);
```

Reply:
Oh, but also, city council districts are only of length one or two. So, we'll need to change the "missingId" from "102" to something like "10"

Reply:
Apologies, I put this comment on the wrong test. This applies to the 404 case later on